src/main/java/com/teka/domain/user/Email.java

```diff
@@ -0,0 +1,16 @@
+package com.teka.domain.user;
+
+import com.teka.domain.user.exception.EmailFormatMismatchException;
+
+import java.util.regex.Pattern;
+
+public record Email(String value) {
+    public Email(String value) {
+        Pattern pattern = Pattern.compile("^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$");
+
+        if (!pattern.matcher(value).matches()) {
+            throw new EmailFormatMismatchException();
+        }
+        this.value = value;
+    }
```

Comment:
```suggestion
    public Email {
        Pattern pattern = Pattern.compile("^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,}$");

        if (!pattern.matcher(value).matches()) {
            throw new EmailFormatMismatchException();
        }
    }
```
P2) record에서는 이렇게 파라미터 목록과 `this.value = value;`와 같은 할당 코드를 생략할 수 있는 압축 생성자가 있습니다. [Oracle 공식 페이지](https://docs.oracle.com/en/java/javase/14/language/records.html)에 관련 설명도 있으므로 참고하고 PhoneNumber에도 적용해주세요.

Reply:
Compact 생성자를 사용하도록 변경했습니다.